pkg/diagnostics/cluster/aggregated_logging/diagnostic.go

```diff
@@ -0,0 +1,173 @@
+	d.result.Info(id, message)
+}
+
+func (d *AggregatedLogging) Error(id string, err error, message string) {
+	d.result.Error(id, err, message)
+}
+
+func (d *AggregatedLogging) Debug(id string, message string) {
+	d.result.Debug(id, message)
+}
+
+func (d *AggregatedLogging) Warn(id string, err error, message string) {
+	d.result.Warn(id, err, message)
+}
+
+func (d *AggregatedLogging) Name() string {
+	return AggregatedLoggingName
+}
+
+func (d *AggregatedLogging) Description() string {
+	return "Check aggregated logging integration for proper configuration"
+}
+
+func (d *AggregatedLogging) CanRun() (bool, error) {
+	if len(d.MasterConfigFile) == 0 {
+		return false, errors.New("No master config file was provided")
+	}
+	if d.OsClient == nil {
+		return false, errors.New("Config must include a cluster-admin context to run this diagnostic")
+	}
+	if d.KubeClient == nil {
+		return false, errors.New("Config must include a cluster-admin context to run this diagnostic")
+	}
+	var err error
+	d.masterConfig, err = hostdiag.GetMasterConfig(d.result, d.MasterConfigFile)
+	if err != nil {
+		return false, errors.New("Unreadable master config; skipping this diagnostic.")
+	}
+	return true, nil
+}
+
+func (d *AggregatedLogging) Check() types.DiagnosticResult {
+	project := retrieveLoggingProject(d.result, d.masterConfig, d.OsClient)
+	if len(project) != 0 {
+		checkServiceAccounts(d, d, project)
+		checkClusterRoleBindings(d, d, project)
+		checkSccs(d, d, project)
+		checkDeploymentConfigs(d, d, project)
+		checkDaemonSets(d, d, project)
+		checkServices(d, d, project)
+		checkRoutes(d, d, project)
+		checkKibana(d.result, d.OsClient, d.KubeClient, project)
+	}
+	return d.result
+}
+
+func retrieveLoggingProject(r types.DiagnosticResult, masterCfg *configapi.MasterConfig, osClient *client.Client) string {
+	r.Debug("AGL0010", fmt.Sprintf("masterConfig.AssetConfig.LoggingPublicURL: '%s'", masterCfg.AssetConfig.LoggingPublicURL))
+
+	loggingUrl, err := url.Parse(masterCfg.AssetConfig.LoggingPublicURL)
+	if err != nil {
+		r.Error("AGL0011", err, "Unable to parse the loggingPublicURL from the masterConfig")
+		return ""
+	}
+
+	routeList, err := osClient.Routes(kapi.NamespaceAll).List(kapi.ListOptions{LabelSelector: loggingSelector.AsSelector()})
+	if err != nil {
+		r.Error("AGL0012", err, fmt.Sprintf("There was an error while trying to find the route associated with '%s'", loggingUrl))
+		return ""
+	}
+
+	for _, route := range routeList.Items {
+		r.Debug("AGL0013", fmt.Sprintf("Comparing URL to route.Spec.Host: %s", route.Spec.Host))
+		if loggingUrl.Host == route.Spec.Host {
+			project := route.ObjectMeta.Namespace
+			r.Debug("AGL0015", fmt.Sprintf("aggregated logging project name: '%s'", project))
+			return project
+		}
+	}
+	r.Error("AGL0014", errors.New("aggregated logging project name is empty"), "Unable to determine the project from the loggingPublicURL defined in the master config")
```

Comment:
Unable to find a route matching the loggingPublicURL defined in the master config:
      [URL]
    Check that the URL is correct and aggregated logging is deployed.

(This may be the only message they see so make it informative)